pkg/clusteragent/autoscaling/workload/model/pod_autoscaler.go

```diff
@@ -0,0 +1,244 @@
+	ScalingValuesVersion *uint64
+
+	// ScalingValuesTimestamp is the time when the values were last updated
+	ScalingValuesTimestamp time.Time
+
+	// HorizontalLastAction is the last horizontal action successfully taken
+	HorizontalLastAction *datadoghq.DatadogPodAutoscalerHorizontalAction
+
+	// HorizontalLastActionError is the last error encountered on horizontal scaling
+	HorizontalLastActionError error
+
+	// VerticalLastAction is the last action taken by the Vertical Pod Autoscaler
+	VerticalLastAction *datadoghq.DatadogPodAutoscalerVerticalAction
+
+	// VerticalRolloutError is the last error encountered on vertical scaling
+	VerticalRolloutError error
+
+	// CurrentReplicas is the current number of PODs for the targetRef
+	CurrentReplicas *int32
+
+	// ScaledReplicas is the current number of PODs for the targetRef matching the resources recommendations
+	ScaledReplicas *int32
+
+	// GlobalError is the an error encountered by the controller not specific to a scaling action
+	GlobalError error
+
+	// Deleted flags the PodAutoscaler as deleted (removal to be handled by the controller)
+	Deleted bool
+}
+
+// NewPodAutoscalerInternal creates a new PodAutoscalerInternal from a Kubernetes CR
+func NewPodAutoscalerInternal(_ string, podAutoscaler *datadoghq.DatadogPodAutoscaler) PodAutoscalerInternal {
+	pai := PodAutoscalerInternal{
+		Namespace:  podAutoscaler.Namespace,
+		Name:       podAutoscaler.Name,
+		Generation: podAutoscaler.Generation,
+		Spec:       podAutoscaler.Spec,
+	}
+	pai.UpdateFromStatus(podAutoscaler)
+	return pai
+}
+
+// ID returns the functional identifier of the PodAutoscaler
+func (p *PodAutoscalerInternal) ID() string {
+	return p.Namespace + "/" + p.Name
+}
+
+// UpdateFromSpec updates the PodAutoscalerInternal from a new spec
+func (p *PodAutoscalerInternal) UpdateFromSpec(podAutoscalerSpec *datadoghq.DatadogPodAutoscalerSpec) {
+	podAutoscalerSpec.DeepCopyInto(&p.Spec)
+	// Reset the target GVK as it might have changed
+	p.TargetGVK = schema.GroupVersionKind{}
+}
+
+// UpdateFromSettings updates the PodAutoscalerInternal from a new settings
+func (p *PodAutoscalerInternal) UpdateFromSettings(podAutoscalerSpec *datadoghq.DatadogPodAutoscalerSpec, settingsVersion uint64, settingsTimestamp time.Time) {
+	p.SettingsVersion = pointer.Ptr(settingsVersion)
+	p.SettingsTimestamp = settingsTimestamp
+
+	podAutoscalerSpec.DeepCopyInto(&p.Spec)
+	podAutoscalerSpec.RemoteVersion = settingsVersion
+
+	// Reset the target GVK as it might have changed
+	p.TargetGVK = schema.GroupVersionKind{}
+}
+
+// UpdateFromStatus updates the PodAutoscalerInternal from an existing status.
+func (p *PodAutoscalerInternal) UpdateFromStatus(podAutoscaler *datadoghq.DatadogPodAutoscaler) {
+	if podAutoscaler.Status.RecommendationsVersion != nil {
+		p.ScalingValuesVersion = pointer.Ptr(*podAutoscaler.Status.RecommendationsVersion)
+		p.ScalingValuesTimestamp = podAutoscaler.Status.UpdateTime.Time
```

Comment:
Should we check for nil `podAutoscaler.Status.UpdateTime` ?

Reply:
Should be impossible but added just in case.